### ISSUE TITLE
Normalize gradient shape function with respect to number of samples in the leaf

### ISSUE THREAD

**Issue by johannesotterbachLDC** [jotterbach/dstk#16]
The `sklearn.DecisionTreeRegressor` contains arrays `tree_.n_node_samples` and `tree_.weighted_n_node_samples` that can be used to reweight the leaf value to be the average rather than the weighted average. The latter leads to large spikes in the generated shape if only few samples are in the final leaf.
